src/EPR.SubsidiaryBulkUpload.Application/Resilience/Pipelines.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Net;
+using EPR.SubsidiaryBulkUpload.Application.Extensions;
+using EPR.SubsidiaryBulkUpload.Application.Options;
+using EPR.SubsidiaryBulkUpload.Application.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Http.Resilience;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Polly.Retry;
+using Polly.Timeout;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Resilience;
+
+public static class Pipelines
+{
+    public const string CompaniesHouseResiliencePipelineKey = "CompaniesHouseResiliencePipeline";
+
+    public static IHttpResiliencePipelineBuilder AddCompaniesHouseResilienceHandler(this IHttpClientBuilder builder) =>
+        builder.AddResilienceHandler(CompaniesHouseResiliencePipelineKey, ConfigureCompaniesHouseResilienceHandler<CompaniesHouseLookupService>());
+
+    public static IHttpClientBuilder AddCompaniesHouseResilienceHandlerToHttpClientBuilder(this IHttpClientBuilder builder)
+    {
+        builder.AddCompaniesHouseResilienceHandler();
+        return builder;
+    }
+
+    public static Action<ResiliencePipelineBuilder<HttpResponseMessage>, ResilienceHandlerContext> ConfigureCompaniesHouseResilienceHandler<T>()
+    {
+        return (builder, context) =>
+        {
+            var apiOptions = context.GetOptions<ApiOptions>();
+
+            builder
+            .AddRetry(new HttpRetryStrategyOptions
+            {
+                Delay = apiOptions.ConvertToTimespan(apiOptions.RetryPolicyInitialWaitTime),
+                ShouldHandle = new PredicateBuilder<HttpResponseMessage>()
+                    .HandleResult(response => response.StatusCode != HttpStatusCode.TooManyRequests),
+                BackoffType = DelayBackoffType.Exponential,
+                MaxRetryAttempts = apiOptions.RetryPolicyMaxRetries,
```

Comment:
I was wondering if there should be two different policy settings for the max retries. 
Only because the max retries on a too many could be substantially different to that of other service errors.

Reply:
I have added a second setting for this, and added it to the config project